tensorflow/core/grappler/optimizers/layout_optimizer.cc

```diff
@@ -226,6 +227,39 @@ bool IsNodeType(const string& node_name, const string& type) {
   return EndWith(node_name, suffix);
 }
 
+bool IsUserDefinedTranspose(const NodeDef* node, const NodeMap* node_map,
+                            const std::vector<int>& perm) {
+  if (node->op() == "Transpose" && node->input_size() == 2) {
```

Comment:
If this Transpose node is a fanin of other nodes in the graph, you can't delete it. You can check this with NodeMap::GetOutputs. Could you add two test cases, one for this "not cacenllable" case (is a fanin for other nodes) and one for the cancelable case (is not fanin for other nodes)?

Reply:
Tests could be added here: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/grappler/layout_optimizer_test.py